src/index.ts

```diff
@@ -133,11 +133,12 @@ export function createStaleWhileRevalidateCache(
           storage.getItem(timeKey),
         ])
 
-        cachedValue = deserialize(cachedValue)
-
         if (isNil(cachedValue) || isNil(cachedAt)) {
```

Comment:
Since `isNil` checks for both `null` and `undefined` we should probably add an extra test to cover for both cases. The mocked storage in the current test returns null, in such case the `cacheGetFailed` event does not get emitted.

Reply:
Well, guess it's not needed if we return before trying to deserialize, but wouldn't hurt with an extra test to make sure we never try to `JSON.parse(undefined)`.

Reply:
It would be beneficial to have a similar cache miss test using a custom storage object that returns `undefined` instead of `null`, like a plain JavaScript object.

Out of interest, what are you using as storage for this use to have popped up?

Reply:
Redis via [keyv](https://github.com/jaredwray/keyv).